src/net.rs

```diff
@@ -132,6 +161,24 @@ pub enum Command {
     // Statistics,
 }
 
+/// read sent bytes, extract the kind of command
+pub fn read_commands<R : Read + Write>(stream : &mut R) -> Result<Command, NetworkErrors> {
+    
+    // read the first byte for code numeric value
+    let status = try!(stream.read_u8());
+    if status != Cnv::CommandPkg as u8 {
+        //send error_package
+        return Err(NetworkErrors::UnknownCmd("command not known".into()))    
+    }
+    
+    // second  4 bytes is the kind of command
+    let command_decode = decode_from(stream, SizeLimit::Bounded(4096));
+    match command_decode {
+        Ok(command) => Ok(command),
+        _ => Err(NetworkErrors::IoError(io::Error::new(ErrorKind::Other, "problems while reading command data")))
+    }
```

Comment:
Instead of that match block, put `decode_from` into `try!()`. So, just:
```rust
// second  4 bytes is the kind of command
Ok(try!(decode_from(stream, SizeLimit::Bounded(4096)))
```